### DISCUSSION TITLE
Are plugin IDs guaranteed to be unique?

### DISCUSSION THREAD

**Discussion by jaxter184** [free-audio/clap#127] [Q&A] [answered]
Are there any guarantees or recommendations by the API that no two CLAP plugins have the same ID? For example by requiring/recommending that all IDs be tied to a URL or something. Of course, there isn't really a good way to ensure this is true, but to phrase my question another way, is it "correct" for a host to assume that these IDs are unique?

https://github.com/free-audio/clap/blob/e60671bec244f794e52c4d2c13a374c79883c51b/include/clap/plugin.h#L17

Also, as a follow-up, the readme mentions that all strings are valid UTF-8. Does this mean that IDs can have any UTF-8 characters in them?

**Comment by baconpaul** [comment, marked as answer]
Most people are using reverse URLs which gives domains. But there is no central registry stopping you from making a plugin like org.surge-synth-team.surge-xt. It would be antisocial but nothing would stop you.

on the second, yes utf-8. If a host doesn’t suppport it it’s a bug. Hosts may have bugs in this regard. We’ve been discussing righting stress test plugins for hosts just like we are writing a plugin validator now.
